sdk/tests/iot/provisioning/test_az_iot_provisioning_client_parser.c

```diff
@@ -369,7 +369,7 @@ test_az_iot_provisioning_client_received_topic_and_payload_parse_invalid_result_
   az_iot_provisioning_client_register_response response;
   az_result ret = az_iot_provisioning_client_parse_received_topic_and_payload(
       &client, received_topic, received_payload, &response);
-  assert_int_equal(AZ_ERROR_ITEM_NOT_FOUND, ret);
+  assert_int_equal(AZ_ERROR_EOF, ret);
```

Comment:
I believe `AZ_ERROR_EOF` makes more sense here given the payload is an incomplete JSON. Please let me know if this needs to be `AZ_ERROR_ITEM_NOT_FOUND` for some reason.

Was this the intention of the test, or should a complete JSON payload be provided (but one that doesn't have the properties we are a looking for).

cc @CIPop, @danewalton

Reply:
do you want to rename EOF to "UNEXPECTED_END"? I think it is the better naming.

Reply:
Yep, that is much better.

Reply:
> do you want to rename EOF to "UNEXPECTED_END"? I think it is the better naming.

Deferring that to outside this PR:
https://github.com/Azure/azure-sdk-for-c/issues/894

Reply:
@ahsonkhan the original item not found is the right error: we've scanned for "status" and could not find it. The code has no expectation that the complete JSON is received.
Please revert to the previous behavior and ensure that none of the existing UTs change behaviors or error codes.